BuildCMakeDocsTarget/action.yml

```
@@ -0,0 +1,53 @@
+name: Build CMake Documentation Target
+description: This action builds release documentation via a CMake target, zips the directory up, and modifies or creates a release to include the .zip file
+author: thebhef
+
+inputs:
+  doc-target:
+    description: the name of the CMake target
+    required: True
+  cmake-directory:
+    description: the directory CMake should use to build
+    required: True
+  doc-output:
+    description: the directory containing the output of the CMake build, relative to cmake-directory
+    required: True
+  build-version:
+    description: the version which should be used for the tag and the name of the zip file
+    required: True
+  create-tag:
+    description: whether to create a tag
+    required: True
+  
+runs:
+  using: composite
+  steps:
+    - name: Configure CMake
+      shell: bash
+      # Configure CMake in a 'build' subdirectory. `CMAKE_BUILD_TYPE` is only required if you are using a single-configuration generator such as make.
+      # See https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html?highlight=cmake_build_type
+      run: cmake -B ${{inputs.cmake-directory}}
```

I would make the generator explicit so we don't have to test with multiple:
```suggestion
      run: cmake -B ${{inputs.cmake-directory}} -G Ninja
```